src/GitHub.VisualStudio/Menus/BlameLink.cs

```diff
@@ -0,0 +1,42 @@
+﻿using GitHub.Services;
+using GitHub.VisualStudio.UI;
+using NullGuard;
+using System;
+using System.Windows;
+
+namespace GitHub.VisualStudio.Menus
+{
+    public class BlameLink : LinkMenuBase, IDynamicMenuHandler
+    {
+
+        public BlameLink(IGitHubServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+        }
+
+        public Guid Guid => GuidList.guidContextMenuSet;
+        public int CmdId => PkgCmdIDList.blameCommand;
+
+        public async void Activate([AllowNull]object data = null)
+        {
+            var isgithub = await IsGitHubRepo();
+            if (!isgithub)
+                return;
+
+            var link = await GenerateLink(true);
+            if (link == null)
+                return;
+            var browser = ServiceProvider.TryGetService<IVisualStudioBrowser>();
+            browser?.OpenUrl(link.ToUri());
+
+            await UsageTracker.IncrementOpenInGitHubCount();
+        }
+
+        public bool CanShow()
+        {
+            var githubRepoCheckTask = IsGitHubRepo();
+            return githubRepoCheckTask.Wait(250) ? githubRepoCheckTask.Result : false;
```

Comment:
I haven't taken too close a look at this, but it really depends on how different the behaviors are between the different types of links. Typically, you have a few options.

1. If the differences between these three links are pretty minimal, make a simple `GitHubLink` class that is configurable and changes behavior accordingly.
2. Inheritance: Have a base `GitHubLink` class that all these link classes inherit from. This is a good approach if there's some reason these absolutely need to be different types. Otherwise I'd prefer to use a composition approach.
3. Composition: Create a `GitHubLink` class that accepts some configuration parameters but also accepts a `GitHubLinkCommand` or a `GitHubLinkHandler` class. This class would actually handle the behavior of clicking on the link while `GitHubLink` would be responsible for the presentation of the link. This keeps these concerns nicely separated.